### ISSUE TITLE
Need Help with Custom Merchant Columns

### ISSUE THREAD

**Issue by marvin9257** [fantasycalendar/FoundryVTT-ItemPiles#331]
I'm trying to integrate the new merchant columns feature into the Twodsix system.  My configuration looks like 
```
Hooks.once("item-piles-ready", async function() {
  /// Called once Item Piles is ready to be used
  game.itempiles.API.addSystemIntegration({
    "VERSION": "1.0.51",

    // The actor class type is the type of actor that will be used for the default item pile actor that is created on first item drop.
    "ACTOR_CLASS_TYPE": "traveller",

    // The item quantity attribute is the path to the attribute on items that denote how many of that item that exists
    "ITEM_QUANTITY_ATTRIBUTE": "system.quantity",

    // The item price attribute is the path to the attribute on each item that determine how much it costs
    "ITEM_PRICE_ATTRIBUTE": "system.price",

    // Item filters actively remove items from the item pile inventory UI that users cannot loot, such as spells, feats, and classes
    "ITEM_FILTERS": [
      {
        "path": "type",
        "filters": "skills,trait,spell"
      },
      {
        "path": "name",
        "filters": game.i18n.localize("TWODSIX.Items.Weapon.Unarmed")
      }
    ],

    "PILE_DEFAULTS": {
      "merchantColumns": [{
        "label": "TWODSIX.Actor.Items.TL",
        "path": "system.techLevel",
        "formatting": "{#}",
        "mapping": {}
      }]
    },

    // Item similarities determines how item piles detect similarities and differences in the system
    "ITEM_SIMILARITIES": ["name", "type", "techLevel"],

    // Currencies in item piles is a versatile system that can accept actor attributes (a number field on the actor's sheet) or items (actual items in their inventory)
    // In the case of attributes, the path is relative to the "actor.system"
    // In the case of items, it is recommended you export the item with `.toObject()` and strip out any module data
    "CURRENCIES": [
      {
        type: "attribute",
        name: "Credits",
        img: "systems/twodsix/assets/icons/id-card.svg",
        abbreviation: "Cr {#}",
        data: {
          path: "system.finances.cash"
        },
        primary: true,
        exchangeRate: 1
      }
    ],

    "CURRENCY_DECIMAL_DIGITS": 0.01
  });
});
```

Whenever I create a new merchant, the custom columns don't appear.  If I edit the merchant through the Configure Piles interface, this specification seems to work OK.  Is there something I'm missing? Thanks.

**Comment by Haxxer** [comment]
I'll see what I can do about updating existing item pile actors with the new columns.

**Comment by marvin9257** [comment]
It also happens when I create a new actor/merchant.  Below is the debug.

<img width="767" alt="Screenshot 2023-04-10 at 8 08 57 AM" src="https://user-images.githubusercontent.com/72580196/230907312-25707794-aa34-4e01-bbc9-876fadcdfaa1.png">

**Comment by marvin9257** [comment]
One other thing I noticed.  Mapping has exist as an object.  Otherwise this line (line 76 of merchant-store.js) fails.
`if (!column?.mapping[AProp] || !column?.mapping[BProp]) {`
Maybe this instead?
if (!column?.mapping?.[AProp] || !column?.mapping?.[BProp]) {

**Comment by Haxxer** [comment]
Fixed in 2.6.1, the `PILE_DEFAULTS` are now considered the default state of item piles, meaning the flags will not show up in the item pile actor flags. Only the diff between the defaults and the current state is saved in the flags.
